setup.py

```diff
@@ -31,32 +31,19 @@ def package_files(directory):
     packages=find_packages(exclude=["tests", "tests.*"]),
     package_data={"mlflow": js_files + models_container_server_files + alembic_files},
     install_requires=[
-        "alembic<=1.4.1",
-        # Required
-        "azure-storage-blob",
-        "click>=7.0",
-        "cloudpickle",
-        "databricks-cli>=0.8.7",
-        "requests>=2.17.3",
-        "six>=1.10.0",
-        'waitress; platform_system == "Windows"',
-        'gunicorn; platform_system != "Windows"',
-        "Flask",
-        "numpy",
-        "pandas",
-        "python-dateutil",
-        "protobuf>=3.6.0",
-        "gitpython>=2.1.0",
-        "pyyaml",
-        "querystring_parser",
-        "docker>=4.0.0",
-        "entrypoints",
-        # Pin sqlparse for: https://github.com/mlflow/mlflow/issues/3433
-        "sqlparse>=0.3.1",
-        # Required to run the MLflow server against SQL-backed storage
-        "sqlalchemy<=1.3.13",
-        "gorilla",
```

Comment:
For skinny client to be shipped, we would need to update the setup.py to be configurable. Alongside deployment scripts to leverage the configuration https://github.com/mlflow/mlflow/issues/3579